stdlib/ballerina-config/src/main/java/org/ballerinalang/bcl/parser/BConfigLangListener.java

```diff
@@ -58,8 +60,59 @@ public void exitKeyVal(TomlParser.KeyValContext context) {
     }
 
     @Override
-    public void enterKey(TomlParser.KeyContext context) {
-        currentKey = context.getText();
+    public void enterSimpleKey(TomlParser.SimpleKeyContext context) {
+        if (!isDottedKey) {
+            currentKey = new StringBuilder(context.getText());
+        } else {
+            super.enterSimpleKey(context);
+        }
+    }
+
+    @Override
+    public void enterDottedKey(TomlParser.DottedKeyContext context) {
+        isDottedKey = true;
+    }
+
+    @Override
+    public void enterDotSep(TomlParser.DotSepContext context) {
+        if (isDottedKey) {
+            currentKey = currentKey.append(context.getText());
+        } else {
+            super.enterDotSep(context);
+        }
+    }
+
+    @Override
+    public void enterQuotedKey(TomlParser.QuotedKeyContext context) {
+        if (isDottedKey) {
+            // if this is a dotted key, need to build the current key by appending
+            if (currentKey != null && !currentKey.toString().isEmpty()) {
+                currentKey = currentKey.append(new StringBuilder(context.basicString().basicStringValue().getText()));
+            } else {
+                currentKey = new StringBuilder(context.basicString().basicStringValue().getText());
+            }
+        } else {
+            currentKey = new StringBuilder(context.basicString().basicStringValue().getText());
+        }
+    }
+
+    @Override
+    public void enterUnquotedKey(TomlParser.UnquotedKeyContext context) {
+        if (isDottedKey) {
+            // if this is a dotted key, need to build the current key by appending
+            if (currentKey != null && !currentKey.toString().isEmpty()) {
+                currentKey = currentKey.append(new StringBuilder(context.getText()));
+            } else {
+                currentKey = new StringBuilder(context.getText());
+            }
+        } else {
+            currentKey = new StringBuilder(context.getText());
+        }
+    }
+
+    @Override
+    public void exitDottedKey(TomlParser.DottedKeyContext ctx) {
+        isDottedKey = false;
```

Comment:
We can simplify this logic for the keys can't we? The rules we have to concentrate on are, `key`, `dotSep`, `unquotedKey` and `quotedKey`. The following should work I think:
```
enterKey -> create a new StringBuilder for current key
enterQuotedKey -> append the basicStringValue to current key
enterUnquotedKey -> append the value to current key
enterDotSep -> append the value to current key
```